src/Dfc.CourseDirectory.Database/Pttcd/Views/FindACourseIndex.sql

```diff
@@ -0,0 +1,61 @@
+﻿CREATE VIEW [Pttcd].[FindACourseIndex] WITH SCHEMABINDING
+	AS SELECT
+		-- Id is the CourseRun's ID appended with the Region, if there is one
+		CONVERT(VARCHAR(36), cr.CourseRunId) + CASE WHEN crr.RegionId IS NOT NULL THEN '-' + crr.RegionId ELSE '' END AS Id,
+
+		1 AS OfferingType, -- Course
+
+		c.CourseId,
+		cr.CourseRunId,
+		CAST(CASE WHEN cr.CourseRunStatus = 1 THEN 1 ELSE 0 END AS BIT) AS Live,
+
+		-- Use the max version from Course, Provider and Venue
+		(SELECT MAX(AllVersions.[Version]) FROM (
+			VALUES (c.[Version]), (p.[Version]), (v.[Version])) AS AllVersions([Version])
+		) AS [Version],
+
+		c.UpdatedOn,
+		p.ProviderId,
+		p.ProviderName AS ProviderDisplayName,
+		c.ProviderUkprn,
+		ld.LearnAimRefTitle QualificationCourseTitle,
+		c.LearnAimRef,
+		ld.NotionalNVQLevelv2,
+		c.CourseDescription,
+		cr.CourseName,
+		cr.DeliveryMode,
+		cr.FlexibleStartDate,
+		cr.StartDate,
+		cr.Cost,
+		cr.CostDescription,
+		cr.DurationUnit,
+		cr.DurationValue,
+		cr.StudyMode,
+		CASE WHEN cr.DeliveryMode = 1 THEN cr.AttendancePattern ELSE null END AttendancePattern,
+		cr.[National],
+		v.VenueName,
+
+		CASE WHEN ISNULL(v.AddressLine1, '') <> '' THEN v.AddressLine1 + ', ' ELSE '' END +
+			CASE WHEN ISNULL(v.AddressLine2, '') <> '' THEN v.AddressLine2 + ', ' ELSE '' END +
+			CASE WHEN ISNULL(v.Town, '') <> '' THEN v.Town + ', ' ELSE '' END +
+			CASE WHEN ISNULL(v.County, '') <> '' THEN v.County + ', ' ELSE '' END +
```

Comment:
https://docs.microsoft.com/en-us/sql/t-sql/language-elements/coalesce-transact-sql?view=sql-server-ver15

`COALESCE(v.County, '')` etc
shorter way of converting nulls to zero length strings

Reply:
I can't see how `COALESCE` is shorter than `ISNULL` here 😕

Reply:
Ah I'd misunderstood what it was doing.
How about `concat_ws`? https://docs.microsoft.com/en-us/sql/t-sql/functions/concat-ws-transact-sql?view=sql-server-ver15

Reply:
hmm, no that's only for nulls, oh well never mind. Who'd use T-SQL anyway, I wish someone would layer on a better language syntax.

Reply:
😍 `concat_ws` is perfect here - even ignores `null`s